netcore/NetCore.Session/NetCoreSessionState.cs

```diff
@@ -0,0 +1,377 @@
+        }
+        public DateTime LastAccessTimeUtc
+        {
+            get { return _lastAccessTimeUtc; }
+            set { _lastAccessTimeUtc = value; }
+        }
+
+        public DateTime ExpirationTimeUtc
+        {
+            get { return _expirationTimeUtc; }
+            set { _expirationTimeUtc = value; }
+        }
+
+        public TimeSpan SpanUntilStale
+        {
+            get { return _spanUntilStale; }
+            set { _spanUntilStale = value; }
+        }
+
+        public byte[] ToByteArray()
+        {
+            int neededBytes = 3 * sizeof(long) + _value.Length;
+            byte[] byteArray = new byte[neededBytes];
+            int byteIndex = 0;
+
+            // Append LastAccessTimeUtc
+            Array.Copy(BitConverter.GetBytes(LastAccessTimeUtc.Ticks), 0, byteArray, byteIndex, sizeof(long));
+            byteIndex += sizeof(long);
+
+            // Append ExpirationTimeUtc
+            Array.Copy(BitConverter.GetBytes(ExpirationTimeUtc.Ticks), 0, byteArray, byteIndex, sizeof(long));
+            byteIndex += sizeof(long);
+
+            // Append SpanUntilStale
+            Array.Copy(BitConverter.GetBytes(SpanUntilStale.Ticks), 0, byteArray, byteIndex, sizeof(long));
+            byteIndex += sizeof(long);
+
+            // Append the value
+            Array.Copy(_value, 0, byteArray, byteIndex, _value.Length);
+            return byteArray;
+        }
+
+        public void FromByteArray(byte[] data)
+        {
+            int byteIndex = 0;
+
+            // Extract the LastAccessTimeUtc
+            LastAccessTimeUtc = DateTime.FromBinary(BitConverter.ToInt64(data, byteIndex));
+            byteIndex += sizeof(long);
+
+            // Extract the ExpirationTimeUtc
+            ExpirationTimeUtc = DateTime.FromBinary(BitConverter.ToInt64(data, byteIndex));
+            byteIndex += sizeof(long);
+
+            // Extract the SpanUntilStale
+            SpanUntilStale = TimeSpan.FromTicks(BitConverter.ToInt64(data, byteIndex));
+            byteIndex += sizeof(long);
+
+            // Extract the value
+            Value = new byte[data.Length - byteIndex];
+            Array.Copy(data, byteIndex, _value, 0, data.Length - byteIndex);
+        }
+    }
+
+    public class GeodeSessionStateCache : GeodeNativeObject, IDistributedCache
+    {
+        private readonly IGeodeCache _cache;
+        private ILogger<GeodeSessionStateCache> _logger;
+        private static Region _region;
+        private string _regionName;
+        private readonly SemaphoreSlim _connectLock = new SemaphoreSlim(initialCount: 1, maxCount: 1);
+
+        public GeodeSessionStateCache(IOptions<GeodeSessionStateCacheOptions> optionsAccessor) {
+
+            var host = optionsAccessor.Value.Host;
+            var port = optionsAccessor.Value.Port;
+            _regionName = optionsAccessor.Value.RegionName;
+
+            _cache = CacheFactory.Create()
+                .SetProperty("log-level", "none")
```

Comment:
Why not allow logging level to be configured?

Reply:
This is a _great_ suggestion - let's externalize log-level for the session, it'll pay us back later.

Reply:
Agreed.